clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -338,6 +338,21 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     if (type_is_instr(memref.instr.type) ||
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
+        bool expect_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
+        std::unique_ptr<instr_t> cur_instr_decoded = nullptr;
+        app_pc next_pc = nullptr;
+        if (expect_encoding) {
+            cur_instr_decoded = std::unique_ptr<instr_t>(new instr_t);
+            instr_init(GLOBAL_DCONTEXT, cur_instr_decoded.get());
+            const app_pc decode_pc = const_cast<app_pc>(memref.instr.encoding);
+            next_pc = decode_from_copy(GLOBAL_DCONTEXT, decode_pc,
```

Comment:
nit: decode_pc can be inlined since it is used only once.

Reply:
Done.